From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joel Wiramu Pauling <aenertia@aenertia.net>
Date: Tue, 18 Mar 2026 00:00:00 +1300
Subject: [PATCH] arm64: dts: rockchip: rk356x: add SLPPIN pinctrl states for
 RK817 PMIC suspend/poweroff

Add GPIO0_PA2 (SLPPIN) pinctrl state definitions to rk3566-anbernic-rgxx3.dtsi
and rk3566-powkiddy-x55.dts so that the lazy pinctrl acquisition in
rk8xx-core.c can coordinate SoC-side pin mux changes with PMIC register
writes during suspend and poweroff.

Boards that already have these states (rk3566-powkiddy-rk2023.dtsi,
rk3568-anbernic-rg-ds.dts) are unaffected.

Three named pinctrl states are declared on the PMIC node:
  - "default"       (pinctrl-0): pmic_int_l only, no SLPPIN mux
  - "pmic-sleep"    (pinctrl-1): PA2 func=1 (PMU_SLEEP) for suspend
  - "pmic-power-off"(pinctrl-2): PA2 GPIO output_low for poweroff

This matches the Rockchip BSP convention for RK3566/RK3568 boards
with RK817/RK809 PMICs.

Signed-off-by: Joel Wiramu Pauling <aenertia@aenertia.net>
---
--- a/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi	2026-03-18 10:39:57.057039666 +1300
+++ b/arch/arm64/boot/dts/rockchip/rk3566-anbernic-rgxx3.dtsi	2026-03-18 10:40:54.189163762 +1300
@@ -264,8 +264,10 @@
 		assigned-clock-parents = <&cru CLK_I2S1_8CH_TX>;
 		#clock-cells = <1>;
 		#sound-dai-cells = <0>;
-		pinctrl-names = "default";
+		pinctrl-names = "default", "pmic-sleep", "pmic-power-off";
 		pinctrl-0 = <&i2s1m0_mclk>, <&pmic_int_l>;
+		pinctrl-1 = <&soc_slppin_slp>;
+		pinctrl-2 = <&soc_slppin_gpio>;
 		wakeup-source;
 
 		vcc1-supply = <&vcc_sys>;
@@ -534,6 +536,19 @@
 			rockchip,pins =
 				<0 RK_PA3 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
+
+		/* GPIO0_PA2 = PMIC SLPPIN mux states for suspend/resume/poweroff */
+		soc_slppin_slp: soc-slppin-slp {
+			rockchip,pins = <0 RK_PA2 1 &pcfg_pull_none>;
+		};
+
+		soc_slppin_gpio: soc-slppin-gpio {
+			rockchip,pins = <0 RK_PA2 RK_FUNC_GPIO &pcfg_output_low>;
+		};
+
+		soc_slppin_gpio_idle: soc-slppin-gpio-idle {
+			rockchip,pins = <0 RK_PA2 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
 	};
 
 	sdio-pwrseq {
--- a/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-x55.dts	2026-03-18 10:39:57.058083412 +1300
+++ b/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-x55.dts	2026-03-18 10:41:12.434336497 +1300
@@ -455,8 +455,10 @@
 		clocks = <&cru I2S1_MCLKOUT_TX>;
 		interrupt-parent = <&gpio0>;
 		interrupts = <RK_PA3 IRQ_TYPE_LEVEL_LOW>;
-		pinctrl-names = "default";
+		pinctrl-names = "default", "pmic-sleep", "pmic-power-off";
 		pinctrl-0 = <&i2s1m0_mclk>, <&pmic_int_l>;
+		pinctrl-1 = <&soc_slppin_slp>;
+		pinctrl-2 = <&soc_slppin_gpio>;
 		wakeup-source;
 		#clock-cells = <1>;
 		#sound-dai-cells = <0>;
@@ -749,6 +751,19 @@
 			rockchip,pins =
 				<0 RK_PA3 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
+
+		/* GPIO0_PA2 = PMIC SLPPIN mux states for suspend/resume/poweroff */
+		soc_slppin_slp: soc-slppin-slp {
+			rockchip,pins = <0 RK_PA2 1 &pcfg_pull_none>;
+		};
+
+		soc_slppin_gpio: soc-slppin-gpio {
+			rockchip,pins = <0 RK_PA2 RK_FUNC_GPIO &pcfg_output_low>;
+		};
+
+		soc_slppin_gpio_idle: soc-slppin-gpio-idle {
+			rockchip,pins = <0 RK_PA2 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
 	};
 
 	sdio-pwrseq {
